torchgeo/datasets/astergdem.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Aster Global Digital Evaluation Model dataset."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, Optional
+
+import matplotlib.pyplot as plt
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+
+
+class AsterGDEM(RasterDataset):
+    """Aster Global Digital Evaluation Model Dataset.
+
+    The `Aster Global Digital Evaluation Model
+    <https://lpdaac.usgs.gov/products/astgtmv003/>`_
+    dataset is a Digital Elevation Model (DEM) on a global scale.
+    The dataset can be downloaded from the
+    `Earth Data website <https://search.earthdata.nasa.gov/search/>`_
+    after making an account.
+
+    Dataset features:
+
+    * DEMs at 30 m per pixel spatial resolution (3601x3601 px)
+    * data collected from the `Aster
+      <https://terra.nasa.gov/about/terra-instruments/aster>`_ instrument
+
+    Dataset format:
+
+    * DEMs are single-channel tif files
+
+    .. versionadded:: 0.3
+    """
+
+    is_image = False
+    filename_glob = "ASTGTMV003_*_dem*"
+    filename_regex = r"""(
+        (?P<name>[ASTGTMV003]{10})
+        _(?P<id>[A-Z0-9]{7})
+        _(?P<data>[a-z]{3})*)"""
```

Comment:
```suggestion
    filename_regex = r"""
        (?P<name>[ASTGTMV003]{10})
        _(?P<id>[A-Z0-9]{7})
        _(?P<data>[a-z]{3})*
    """
```
I don't think this outer set of parentheses is needed. Also, I would put the ending quotes on a new line so that indentation is clear, same as we do for docstrings.